system/xmllib/xpathprocessor.hpp

```diff
@@ -88,9 +88,9 @@ class CXpathContextScope : CInterface
     Linked<IProperties> namespaces;
 public:
     IMPLEMENT_IINTERFACE;
-    CXpathContextScope(IXpathContext *ctx, const char *name, IProperties *ns=nullptr) : context(ctx), namespaces(ns)
+    CXpathContextScope(IXpathContext *ctx, const char *name, bool templateChildScope, IProperties *ns=nullptr) : context(ctx), namespaces(ns)
     {
-        context->beginScope(name);
+        context->beginScope(name, templateChildScope);
```

Comment:
Should the potentially NULL context be blindly dereferenced, or would a NULL check with exception be preferred?

Personally, my interfaces tend to accept references instead of pointers if I want to assume receipt of a valid instance.